egs/wsj/s5/steps/nnet3/chain/train.py

```diff
@@ -259,6 +259,10 @@ def train(args, run_opts, background_process_handler):
     num_jobs = common_lib.get_number_of_jobs(args.tree_dir)
     feat_dim = common_lib.get_feat_dim(args.feat_dir)
     ivector_dim = common_lib.get_ivector_dim(args.online_ivector_dir)
+    num_cmn_offsets = train_lib.get_num_cmn_offsets(args.feat_dir)
```

Comment:
... actually, I'd like to suggest a change to the way the iVectors are stored in the online-ivector dir (this is a script change).
My concern is that with the current setup we'll have problems later on, in lattice generation for discriminative training, if we leave the iVectors with the higher-than-normal dimension.  We can't really afford to generate the discriminative-training lattices for all those offsets, so we probably don't want to integrate this stuff with that, at least for now.
What I think would be better is to have one archive, ivector_online.ark, but two scp files:
`ivector_online_offset.scp` which is generated along with ivector_online.ark, and contains the higher-dimensional iVectors with offsets, and `ivector_online.scp`, which contains just the "no-offset" version of each set of iVector [which I believe is offset-index zero].  E.g. if ivector_online_offset.scp has a line like `some_spk /foo/bar/ivector_online.ark:54321`, ivector_online.scp will have a line like: `some_spk /foo/bar/ivector_online.ark:54321[,0:99]` containing a column range of that matrix.